File: Data Structures & Algorithms/maximum-subarray/submission-0.cpp
class Solution{
public:
    int maxSubArray(vector<int>& nums){
        int currentSum=nums[0];
        int best=nums[0];
        for(int i=1;i<nums.size();i++){
            currentSum=max(nums[i],currentSum+nums[i]);
            best=max(best,currentSum);
        }
        return best;
    }
};

File: Data Structures & Algorithms/meeting-schedule-ii/submission-0.java
class Solution{
    public int minMeetingRooms(List<Interval> intervals){
        int[] start=new int[intervals.size()];
        int[] end=new int[intervals.size()];
        for(int i=0;i<intervals.size();i++){
            start[i]=intervals.get(i).start;
            end[i]=intervals.get(i).end;
        }
        Arrays.sort(start);
        Arrays.sort(end);
        int s=0;
        int e=0;
        int currRooms=0;
        int best=0;
        while(s<intervals.size()){
            if(start[s]<end[e]){
                currRooms++;
                s++;
            }else{
                currRooms--;
                e++;
            }
            best=Math.max(best,currRooms);
        }
        return best;
    }
}